scripts/build.py

```diff
@@ -1257,25 +1288,38 @@ def generate_build_directory(hashes):
 def main(args=None):
     """The main method of this script."""
     options = _PARSER.parse_args(args=args)
+
+    if options.maintenance_mode and not options.prod_env:
+        raise Exception(
+            'maintenance_mode should only be enabled in prod build.')
+
     # Regenerate /third_party/generated from scratch.
     safe_delete_directory_tree(THIRD_PARTY_GENERATED_DEV_DIR)
     build_third_party_libs(THIRD_PARTY_GENERATED_DEV_DIR)
 
     # If minify_third_party_libs_only is set to True, skips the rest of the
     # build process once third party libs are minified.
-    if options.minify_third_party_libs_only and not options.prod_env:
-        raise Exception(
-            'minify_third_party_libs_only should not be set in non-prod env.')
+    if options.minify_third_party_libs_only:
+        if options.prod_env:
+            minify_third_party_libs(THIRD_PARTY_GENERATED_DEV_DIR)
+            return
+        else:
+            raise Exception(
+                'minify_third_party_libs_only should not be '
+                'set in non-prod env.')
+
+    modify_constants(options.prod_env, options.maintenance_mode)
     if options.prod_env:
         minify_third_party_libs(THIRD_PARTY_GENERATED_DEV_DIR)
-        if not options.minify_third_party_libs_only:
-            hashes = generate_hashes()
-            if options.deparallelize_terser:
-                build_using_webpack(WEBPACK_TERSER_CONFIG)
-            else:
-                build_using_webpack(WEBPACK_PROD_CONFIG)
-            generate_app_yaml(deploy_mode=options.deploy_mode)
-            generate_build_directory(hashes)
+        hashes = generate_hashes()
```

Comment:
The removal of "if not options.minify_third_party_libs_only" from here seems to change the behaviour, since now hashes are always computed. Why do we do this? Or is the point that the line before this should be removed? I'm somewhat confused...